src/layers.rs

```diff
@@ -74,6 +77,7 @@ impl Layer {
             ],
             required: [
                 ("name", name, |v| Some(v)),
+                ("id", id, |v:String| v.parse().ok()),
             ],
             TiledError::MalformedAttributes("layer must have a name".to_string())
```

Comment:
Seems like we need to either update this string or don't require the `id` attribute. I don't personally see why the IDs should be required, so I'd suggest moving it to `optionals`.

Reply:
I was under the impression that layers are required to have the `id` attribute since if you edit the file to remove it (or set it to something invalid), Tiled adds one back.

Reply:
Hmm, yeah layers will always get an ID assigned, but there is just no reason to fail loading a map just because there is no `id` attribute.

Reply:
Fair. How should that be handled though, store `id` as `Option<u32>` or keep it raw and assign a default value?
We could use 0 since it's normally unused, or we could base it on `nextlayerid` like Tiled (it might be weird to have hidden functionality like that though)

Reply:
I think it should stay on 0. Indeed layers would never get ID 0, so it's a suitable value for when the attribute isn't there, and indeed automatically assigning IDs isn't useful outside of an editor.